src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -33,34 +61,47 @@ export class SavedObjectsClient {
    */
   async bulkCreate(objects, options = {}) {
     const action = options.force === true ? 'index' : 'create';
-
-    const body = objects.reduce((acc, object) => {
-      acc.push({ [action]: { _type: object.type, _id: object.id } });
-      acc.push(object.attributes);
-
+    let response;
+    const v6Body = objects.reduce((acc, object) => {
```

Comment:
We can add an option to define the format to use and default to v6. If it fails, we can re-call bulkCreate with v5 if it's v6.

We can also create a module for each formatting helpers.